flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedSequenceNumberingViewReader.java

```diff
@@ -119,7 +124,7 @@ public boolean isAvailable() {
 			return subpartitionView.isAvailable();
 		}
 		else {
-			return subpartitionView.nextBufferIsEvent();
+			return !subpartitionView.isBlockedByCheckpoint() && subpartitionView.nextBufferIsEvent();
```

Comment:
From `subpartitionView` prospective , it is better to give a overall method for judging its condition, not spreading multiple conditions. we might need to refactor the current `ResultSubpartition#isAvailable` for also considering blocked state with `nextBufferIsEvent`.

Reply:
I unified the ```isBlockedByCheckpoint``` tag with ```isAvailable```.